cmd/juju/commands/migrate.go

```diff
@@ -178,3 +240,162 @@ func (c *migrateCommand) getTargetControllerMacaroons() ([]macaroon.Slice, error
 	defer api.Close()
 	return httpbakery.MacaroonsForURL(jar, api.CookieURL()), nil
 }
+
+func (c *migrateCommand) checkMigrationFeasibility(spec *controller.MigrationSpec) error {
```

Comment:
I had initially thought that we should be doing this feasibility check in the apiserver as part of the "precheck". That way API users of migration would also get this user check rather than just having it in the CLI. Thoughts?

Reply:
Making this work server-side was my initial approach. However, after chatting with @mitechie to better understand the use-case it seems like in the most likely scenario the admin would have the _newer_ version of the cli and would try to migrate a model from an _older_ controller to a (potentially) newer controller.

With that in mind, and given that the migration pre-check happens in the source controller it seemed like a better strategy to implement the feasibility check on the client; all the required APIs are implemented even by older controllers so we could get these checks for free without having to upgrade first.

Reply:
This makes sense for the current release. I still think we should also add to the controller checks so that someone just using the API will get an error. This can be done for 2.6.1 (or whenever), but we shouldn't forget.

Reply:
@howbazaar A follow-up card has been created for adding the server-side logic for this feature. I think I can also land the server-side bits today as the checks are pretty straightforward and should be relatively easy to port.